controllers/QuizRunSubmission.controller.js

```diff
@@ -0,0 +1,85 @@
+const QuizRunSubmission = require("../models/QuizRunSubmission.model.js");
+
+exports.create = (req, res) => {
+  const quizRunSubmission = new QuizRunSubmission({
+    ...req.body,
+    createdBy: req.user,
+  });
+  QuizRunSubmission.findOne({
+    username: req.body.username,
+    quizId: req.body.quizId,
+    runId: req.body.runId,
+    questionNo: req.body.questionNo,
+  })
+    .then((response) => {
+      if (response == null) {
+        quizRunSubmission
+          .save()
+          .then((data) => {
+            res.send(data);
+          })
+          .catch((err) => {
+            res.status(500).send({
+              message:
+                err.message ||
+                "Some error occurred while submiting the answer.",
+            });
+          });
+      } else {
+        return res.status(404).send({
+          message: req.body.username + " has already posted the answer",
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message:
+          err.message || "Some error occurred while adding the submission.",
+      });
+    });
+};
+
+exports.findSelectedOptionsQuestionResults = (req, res) => {
+  const users = {
+    A: [],
+    B: [],
+    C: [],
+    D: [],
+  };
+  QuizRunSubmission.find({
```

Comment:
make a single query for all submission to a question and then find out the total count in the array for each option. Do not make multiple calls for each option